dagger/main.go

```diff
@@ -1,19 +1,43 @@
+// A generated module for Dagger functions
+//
+// This module has been generated via dagger init and serves as a reference to
+// basic module structure as you get started with Dagger.
+//
+// Two functions have been pre-created. You can modify, delete, or add to them,
+// as needed. They demonstrate usage of arguments and return types using simple
+// echo and grep commands. The functions can be called from the dagger CLI or
+// from one of the SDKs.
+//
+// The first line in this comment block is a short description line and the
+// rest is a long description with more detail on the module's purpose or usage,
+// if appropriate. All modules should have a short description.
+
 package main
 
 import (
 	"context"
 	"dagger/dagger/internal/dagger"
-	"fmt"
-	"os"
-	"time"
+	"log"
+	"os/exec"
+	"strings"
 )
 
 type Dagger struct{}
 
-func (m *Dagger) TestChain(ctx context.Context, chain string, image string, source *dagger.Directory) (*dagger.Container, error) {
+func generateRandomHex() (string, error) {
+	cmd := exec.Command("sh", "-c", "cat /dev/urandom | head -c 32 | xxd -p -c100")
+
+	output, err := cmd.Output()
+	if err != nil {
+		return "", err
+	}
+
+	hexString := strings.TrimSpace(string(output))
+	return hexString, nil
+}
```

Comment:
This is a bad way to generate random numbers.  Could do it natively in go instead.

However, we don't really need this to be random.  I suggest deleting and using a fixed random hex PRIVATE_KEY, like I had before.